apps/omg_watcher_info/lib/omg_watcher_info/db/repo.ex

```diff
@@ -33,16 +34,25 @@ defmodule OMG.WatcherInfo.DB.Repo do
   def insert_all_chunked(_schema_or_source, [], _opts), do: :ok
 
   def insert_all_chunked(schema_or_source, entries, opts) do
-    chunk_size = @max_params_count |> div(entries |> hd |> fields_count)
-
     utc_now = DateTime.utc_now()
+    entries = Enum.map(entries, fn entry -> Map.merge(entry, %{inserted_at: utc_now, updated_at: utc_now}) end)
+
+    chunk_size = chunk_size(entries |> hd)
 
     entries
-    |> Enum.map(fn entry -> Map.merge(entry, %{inserted_at: utc_now, updated_at: utc_now}) end)
     |> Stream.chunk_every(chunk_size)
     |> Enum.each(&insert_all(schema_or_source, &1, opts))
   end
 
+  # Note: an entry with 0 fields will cause a divide-by-zero error.
+  #
+  # DB.Repo.chunk_size(%{}) ==> (ArithmeticError) bad argument in arithmetic expression
+  #
+  # Do we want/need to be that defensive?
+  def chunk_size(entry) do
+    @max_params_count |> div(entry |> fields_count)
```

Comment:
```suggestion
    fields_count = fields_count(entry)
    div(@max_params_count, fields_count)
```

I see this is old code but I think it's an overuse of pipes. Maybe this?